pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -0,0 +1,434 @@
+	if count := len(podCNIInfo.NetworkConfig); count == 0 {
+		return defaultSecondaryInterfaceName
+	} else {
+		// Generate new interface name (eth1,eth2..eth10) and return to caller.
+		for ifaceIndex := start_iface_index; ifaceIndex < end_iface_index; ifaceIndex++ {
+			ifName := fmt.Sprintf("%s%d", "eth", ifaceIndex)
+			_, exist := podCNIInfo.NetworkConfig[ifName]
+			if !exist {
+				return ifName
+			}
+		}
+	}
+	// Generates random interface name and return. Above execution will try to ensure allocating interface names in order.
+	// If none available between eth1 to eth10 (already used per Pod), generate random name with the integer range.
+	return string("eth") + strconv.Itoa(rand.IntnRange(end_iface_index, max_rand_index))
+}
+
+func removePodAllSecondaryNetwork(podCNIInfo *cnipodcache.CNIConfigInfo) error {
+	var cmdArgs *invoke.Args
+	// Clean-up IPAM at whereabouts db (etcd or kubernetes API server) for all the secondary networks of the Pod which is getting removed.
+	// NOTE: SR-IOV VF interface clean-up, upon Pod delete will be handled by SR-IOV device plugin. Not handled here.
+	cmdArgs = &invoke.Args{Command: string("DEL"), ContainerID: podCNIInfo.ContainerID,
+		NetNS: podCNIInfo.ContainerNetNS, Path: cniPath}
+	// example: podCNIInfo.NetworkConfig = {"eth1": net1-cniconfig, "eth2": net2-cniconfig}
+	for secNetInstIface, secNetInstConfig := range podCNIInfo.NetworkConfig {
+		cmdArgs.IfName = secNetInstIface
+		// Do DelIPAMSubnetAddress on network config (secNetInstConfig) and command argument (updated with interface name).
+		err := ipam.DelIPAMSubnetAddress(secNetInstConfig, cmdArgs)
+		if err != nil {
+			return fmt.Errorf("Failed to clean-up whereabouts IPAM %v", err)
+		}
+		// Delete map entry for secNetInstIface, secNetInstConfig
+		delete(podCNIInfo.NetworkConfig, secNetInstIface)
+	}
+	return nil
+}
+
+func (pc *PodController) enqueuePod(obj interface{}) {
+	var err error
+	pod, isPod := obj.(*corev1.Pod)
+	if !isPod {
+		podDeletedState, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			klog.ErrorS(err, "Unexpected object received:", obj)
+			return
+		}
+		pod, ok := podDeletedState.Obj.(*corev1.Pod)
+		if !ok {
+			klog.ErrorS(err, "DeletedFinalStateUnknown object is not of type Pod: ", podDeletedState.Obj, pod)
+			return
+		}
+	}
+	podKey := podKeyGet(pod)
+	pc.queue.Add(podKey)
+}
+
+// handleAddUpdatePod handles Pod Add, Update events and updates annotation if required.
+func (pc *PodController) handleAddUpdatePod(obj interface{}) error {
+	var ret bool
+	var err error
+	var podCNIInfo *cnipodcache.CNIConfigInfo
+	pod := obj.(*corev1.Pod)
+	if len(pod.Status.PodIPs) == 0 {
+		// Primary network configuration is not complete yet.
+		// Note: Return nil here to unqueue Pod add event. Secondary network configuration will be handled with Pod update event.
+		return nil
+	}
+	secondaryNetwork, ret := checkForPodSecondaryNetworkAttachement(pod)
```

Comment:
nit: typically the Go convention is to use `ok` for the name of such a variable name, not `ret`